contrib/src/main/scala/akka/stream/contrib/Valve.scala

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.contrib
+
+import akka.NotUsed
+import akka.stream.stage.{GraphStageLogic, GraphStageWithMaterializedValue, InHandler, OutHandler}
+import akka.stream._
+import akka.stream.contrib.SwitchMode.{Close, Open}
+
+import scala.concurrent.{Future, Promise}
+
+/**
+  * Pause/ Resume a Flow
+  */
+trait ValveSwitch {
+  /**
+    * Change the state of the valve
+    *
+    * @param mode expected mode to switch on
+    * @return A future that completes with true if the mode did change and false if it already was in the requested mode
+    */
+  def flip(mode: SwitchMode): Future[Boolean]
+}
+
+
+object Valve {
+  /**
+    * Factory for [[Valve]] instances.
+    *
+    * @tparam A input and output type
+    * @return [[Valve]] Opened instance
+    *
+    */
+  def apply[A](): Valve[A] = new Valve[A](SwitchMode.Open)
+
+  /**
+    * Factory for [[Valve]] instances.
+    *
+    * @tparam A input and output type
+    * @return [[Valve]] instance
```

Comment:
I think these two (@param and @tparam) are superflous, that it returns an instance of `Valve` is obvious from the type signature. 

The factory methods could be made more self-documenting by not typing with the concrete type but with `Graph[FlowShape[A, A], ValveSwitch]` - that way there will not really be a need for any docs except the single line already present.

Also, again the scaladoc format should be a single space indent, like javadocs, so:
```
/**
 * 
 * 
```
and not as it is now:
```
/** 
  *
  *
```